File: drills/ts/solutions/nc26_sliding_window_maximum__brute_force.ts
export function maxSlidingWindow(nums: number[], k: number): number[] {
  if (k <= 0 || nums.length < k) return [];
  const out: number[] = [];
  for (let i = 0; i + k <= nums.length; i++) out.push(Math.max(...nums.slice(i, i + k)));
  return out;
}

File: drills/go/solutions/nc26_sliding_window_maximum__brute_force.go
package main

func maxSlidingWindow(nums []int, k int) []int {
	result := make([]int, 0, len(nums)-k+1)
	for i := 0; i+k <= len(nums); i++ {
		best := nums[i]
		for _, n := range nums[i+1 : i+k] {
			best = max(best, n)
		}
		result = append(result, best)
	}
	return result
}
